crates/core/src/surfnet/locker.rs

```diff
@@ -1506,14 +1506,40 @@ impl SurfnetSvmLocker {
             capture
         };
 
+        // Determine if this is a simulation (no signature) or execution (has signature)
+        // For now, always generate a UUID and store in simulated_transaction_profiles
+        let uuid = uuid::Uuid::new_v4();
+        let mut is_execution = false;
+        // If the transaction has a non-default signature, treat as execution
+        if signature != solana_signature::Signature::default() {
+            is_execution = true;
+        }
+
+        let profile_result = ProfileResult {
+            compute_units: compute_units_estimation_result,
+            state: ProfileState::new(pre_execution_capture, post_execution_capture),
+            slot,
+            uuid: Some(uuid),
+        };
+
+        self.with_svm_writer(|svm| {
+            svm.simulated_transaction_profiles
+                .insert(uuid, profile_result.clone());
+            if is_execution {
+                svm.executed_transaction_profiles
+                    .insert(signature, profile_result.clone());
+                svm.profile_tag_map
+                    .entry(signature.to_string())
+                    .or_default()
+                    .push(UuidOrSignature::Uuid(uuid));
+            }
```

Comment:
This logic seems off. I thought it should be:
1. If simulation we insert into `simulated_transaction_profiles` and _not_ `executed_transaction_profiles`.
2. If not simulation, we insert into `executed_transaction_profiles` and _not_ `simulated_transaction_profiles`.
3. We _always_ insert into `profile_tag_map`